src/foremast/s3/s3apps.py

```diff
@@ -116,14 +116,37 @@ def _set_bucket_dns(self):
         LOG.info("Created DNS %s for Bucket", self.bucket)
 
     def _put_bucket_tagging(self):
-        tagset = {'TagSet': [
-            {'Key': 'app_name',
-             'Value': self.app_name},
-            {'Key': 'app_group',
-             'Value': self.group},
-        ]}
-        self.s3client.put_bucket_tagging(Bucket=self.bucket, Tagging=tagset)
-        LOG.info("Adding tagging %s for Bucket", tagset)
+        """Regular put_bucket_tagging sets TagSet which overwrites old tags.
+        Below logic keeps the old tags in place as well.
+        """
+        app_group_tag = {'Key': 'app_group', 'Value': self.group}
+        app_name_tag = {'Key': 'app_name', 'Value': self.app_name}
+        name_tag_count, group_tag_count = 0, 0
+        try:
+            # Get current tags list, if no tags exist will get an exception.
+            result = self.s3client.get_bucket_tagging(Bucket=self.bucket)['TagSet']
+            for item in result:
```

Comment:
Wouldn't this be simpler to blindly copy all tags found an reinsert them?
The gathering of the tags should be a separate method too.